modular_zubbers/code/datums/components/vore/belly.dm

```diff
@@ -203,6 +221,27 @@
 			var/new_sound = tgui_input_list(usr, "Pick an release sound", "Release Sound", sounds_to_pick_from, release_sound)
 			if(new_sound)
 				release_sound = new_sound
+		// Auto-transfer settings
+		if("autotransfer_enabled")
+			autotransfer_enabled = !autotransfer_enabled
+			if(!autotransfer_enabled)
+				autotransfer_timer = 0
+		if("autotransfer_target")
+			var/list/belly_choices = list()
+			for(var/obj/vore_belly/belly as anything in owner.vore_bellies)
+				if(belly != src) // Don't allow transferring to self
+					belly_choices[belly.name] = belly
+			if(LAZYLEN(belly_choices))
+				var/choice = tgui_input_list(usr, "Select target belly for auto-transfer", "Auto-Transfer Target", belly_choices)
+				if(choice)
+					autotransfer_target = belly_choices[choice]
+			else
+				to_chat(usr, span_warning("You need at least one other belly to set as a transfer target!"))
+		if("autotransfer_delay")
+			var/new_delay = tgui_input_number(usr, "Transfer delay in seconds", "Auto-Transfer Delay", autotransfer_delay / 10, 300, 5)
+			if(new_delay)
+				autotransfer_delay = new_delay * 10 // Convert to deciseconds
+				autotransfer_timer = 0 // Reset timer
```

Comment:
technically this can be modularized since it's just a switch statemen, so you'd have to just create another switch statement in the extended proc, but otherwise just make splurt modularization comments around it